### PR TITLE
feat: add eth fallback doc

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

https://celestiaorg.github.io/docs/pr-preview/pr-1286/developers/eth-fallback

- [x] add functions for optimism and short description
- [x] add functions for arbitrum and short description

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: developers/eth-fallback.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+to store data onchain if the primary data availability storage
+fails.
+
+The [@celestiaorg/nitro](https://github.com/celestiaorg/nitro) integration
+[uses the same fallback mechanism](https://github.com/celestiaorg/nitro/blob/f01968eb3d4e19329e9c92b050e98a8e5772f1f2/arbnode/batch_poster.go#L845-L857).
+
+## OP Stack
+
+The ETH fallback mechanism is set up in the
+[celestiaorg/optimism integration](https://github.com/celestiaorg/optimism).
```
Comment (tuxcanfly) [inline developers/eth-fallback.md:44]:
```suggestion
[celestiaorg/optimism integration](https://github.com/celestiaorg/optimism/tree/tux/rebase-frame-ref-version).
```

File: developers/eth-fallback.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+The `driver.go` and `calldata_source.go` files are part of the ETH fallback
```
Comment (tuxcanfly) [inline developers/eth-fallback.md:46]:
```suggestion
The `op-batcher/batcher/driver.go` and `op-node/rollup/derive/calldata_source.go` files are part of the ETH fallback
```

File: developers/eth-fallback.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+) {
+    // ...
+}
+```
+
+In `calldata_source.go`,
+[the `DataFromEVMTransactions` function defines the read path](https://github.com/celestiaorg/optimism/blob/1215c15fda540a1f19b81588de98e2e7b546e517/op-node/rollup/derive/calldata_source.go#L131-L180)
+of the ETH fallback. This function filters all of the transactions and returns
+the calldata from transactions that are sent to the batch inbox address from
+the batch sender address. It checks the type of the frame and retrieves the
```
Comment (tuxcanfly) [inline developers/eth-fallback.md:72]:
```suggestion
the batch sender address. It checks the type of the frame by reading the version prefix and retrieves the
```

File: developers/eth-fallback.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+mechanism in the op-batcher and op-node respectively.
+
+In [`driver.go`, the `sendTransaction` function is responsible for the write path](https://github.com/celestiaorg/optimism/blob/1215c15fda540a1f19b81588de98e2e7b546e517/op-batcher/batcher/driver.go#L351-L395)
+of the ETH fallback. This function creates and submits a transaction to the
+batch inbox address with the given data. It uses the underlying `txmgr` to
+handle transaction sending and price management. If the transaction data
+can be published to Celestia, it creates a `FrameCelestiaStdRef` and sends
+the transaction with this data. If it cannot be published to Celestia, it
+falls back to Ethereum by creating a `FrameEthereumStdRef` and sends the
+transaction with this data.
```
Comment (tuxcanfly) [inline developers/eth-fallback.md:56]: Would be worth mentioning that the transaction data includes a version prefix, which determines how the data will be parsed.

* `v0 - FrameCelestiaLegacyRef - version prefix 0x00`: legacy celestia  - 8 bytes block height, 4 bytes tx index
* `v1 - FrameEthereumStdRef - version prefix 0x01`: eth calldata fallback - all remaining bytes are interpreted as Frame
* `v2 - FrameCelestiaStdRef - version prefix 0x02`: standard celestia - 8 bytes block height, 32 byte tx commitment

Comment (tuxcanfly) [reply developers/eth-fallback.md:56]: In other words the first byte of the calldata is interpreted as the version prefix which determines how to parse the remaining data.

**Review by tuxcanfly — APPROVED** [review]
Looks good. Added a few suggestions for clarity.

File: developers/eth-fallback.md (hunk 4 above)
Comment (jcstein) [reply developers/eth-fallback.md:57]: https://github.com/celestiaorg/docs/commit/bf54f16d00507331093cde40ea00b91fa9cdf97f

**Pull request merged**